### ISSUE TITLE
Cloudflare Tunnel

### ISSUE THREAD

**Issue by Jzsiba** [IceWhaleTech/CasaOS-AppStore#158]
### App Information

- Name: Cloudflare Tunnel
- Short Description: A simple tool used to expose your home network resources to the web, securely. 
- Official Website: Cloudflare.com
- GitHub Repository: https://github.com/cloudflare/cloudflared
- Docker Image: cloudflare/cloudflared

### Why do you want this app?

Using Cloudflare Tunnel (free), you can point a domain name that you own at your docker containers running on your private casaos instance, and the tunnel traffics network through Cloudflare securely, so you're not actually exposing your local network to the Internet. 

They have an easy interface to set up a domain name and create subdomains. Which you then point at your various containers.

For instance, I am running 2 docker containers for MariaDB and another for chevereto-free (_Both of which I'll also request for Appification_). Here https://img.hitpoints.online/ by creating a subdomain pointed at the port that it's running on (8020) and it looks for all intents and purposes as though it's running like a normal site. It even includes SSL that you don't have to configure through CasaOS in any way.

Similarly I am also running home assist, Uptime Kuma & pihole that I can access in the same fashion. 

### Additional information?

It is not possible to get it running by using the "import" function within CasaOS. The environment configuration is different enough that the import function doesn't appear to work. 

You have to ssh in to your system and run the command there.
example (_fake tunnel ID included_), you must all -d to the command given to you by cloudflare. 
`docker run -d cloudflare/cloudflared:latest tunnel --no-autoupdate run --token o4MQY6TpJIexdUt3pTWmgzcMO2YD1pQlJ0N14w0MG0kwIVMYKkjTlGOFaMTDMkjE4WmyIYliTNVMUNZpW3qTllRciUlOk0ZjZklUdhDmCjMNiIYahxdxOUbI4i0yWgLhWL2dmNjVJE0JTTZ3Zo9njFVYZUObM30zSzMIhAZyADWFmyF05iVtjwD6`

Once you run that, you can import it via the CasaOS interface. I followed this guys tutorial https://www.youtube.com/watch?v=OAeQwdFXsQQ

**Comment by charagender** [comment]
There's no way to do this as of yet - but if my issue gets resolved I'd be able to create an app for this, be on the look-out for any updates on that!

**Comment by WisdomSky** [comment]
I have a docker image currently in the works for cloudflared. 
It has a WebUI where you will input your cloudflare tunnel token. 

![image](https://github.com/IceWhaleTech/CasaOS-AppStore/assets/6792172/9c09c67f-03d4-4f6d-942c-acd30a128d4d)

Clicking the `Install Tunnel Service`, will start the tunnel

**Comment by WisdomSky** [comment]
Cloudflared for CasaOS is now up.

You can install it by using the custom appstore I made:  https://github.com/WisdomSky/Casaos-Appstore

_Note: Requires updating to CasaOS 0.4.4+_

**Comment by WisdomSky** [comment]
#177

**Comment by Jzsiba** [comment]
This is awesome, thanks WisdomSky! I will mark this as solved/closed/whatever github means when you do that thing.

**Comment by yieme** [comment]
Kudo's @WisdomSky, this is awesome! As cloudflared would be a gateway into CasaOS, you might want to look into the HIGH priority security issues identified by [trivy](https://github.com/aquasecurity/trivy) via `trivy i wisdomsky/cloudflared-web`

trivy i wisdomsky/cloudflared-web | grep Total

```
Total: 81 (UNKNOWN: 0, LOW: 67, MEDIUM: 11, HIGH: 3, CRITICAL: 0)
Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)
```

for reference cloudflare's official is:

trivy i cloudflare/cloudflared | grep Total
```
Total: 15 (UNKNOWN: 0, LOW: 11, MEDIUM: 4, HIGH: 0, CRITICAL: 0)
Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 0)
```

**Comment by WisdomSky** [comment]
> you might want to look into the HIGH priority security issues identified by [trivy](https://github.com/aquasecurity/trivy)

I have no plans of explicitly patching these issues manually. I'll just have to leave it to the maintainers of the base image [`library/node`](https://hub.docker.com/_/node/) used by [`wisdomsky/cloudflared-web`](https://hub.docker.com/r/wisdomsky/cloudflared-web) to release an updated image before I'll build a new image.

Note: 2 of the High severity issues (CVE-2023-2953) listed by trivy are from openldap v2.5.X. The issue was patched and fixed in v2.6 but it's still in experimental stage.

**Comment by tigerinus** [comment]
@Jzsiba @yieme @WisdomSky - sounds like this issue can be closed.

If not please re-open
